kubernetes.go

```diff
@@ -58,18 +60,18 @@ func newKubeController(ctx context.Context, c *kubernetes.Clientset, gw *gateway
 	}
 
 	if existGatewayCRDs(ctx, gw) {
-		if resource := lookupResource("HTTPRoute"); resource != nil {
-			gatewayController := cache.NewSharedIndexInformer(
-				&cache.ListWatch{
-					ListFunc:  gatewayLister(ctx, ctrl.gwClient, core.NamespaceAll),
-					WatchFunc: gatewayWatcher(ctx, ctrl.gwClient, core.NamespaceAll),
-				},
-				&gatewayapi_v1beta1.Gateway{},
-				defaultResyncPeriod,
-				cache.Indexers{gatewayUniqueIndex: gatewayIndexFunc},
-			)
-			ctrl.controllers = append(ctrl.controllers, gatewayController)
+		gatewayController := cache.NewSharedIndexInformer(
+			&cache.ListWatch{
+				ListFunc:  gatewayLister(ctx, ctrl.gwClient, core.NamespaceAll),
+				WatchFunc: gatewayWatcher(ctx, ctrl.gwClient, core.NamespaceAll),
+			},
+			&gatewayapi_v1beta1.Gateway{},
+			defaultResyncPeriod,
+			cache.Indexers{gatewayUniqueIndex: gatewayIndexFunc},
+		)
+		ctrl.controllers = append(ctrl.controllers, gatewayController)
```

Comment:
what would happen if you don't have GwAPI CRDs installed? is there a way to make sure we don't crash the pod when a cluster is missing gwAPI CRDs?

Reply:
yeah, that's already handled by the existGatewayCRDs function

Here: [Function](https://github.com/ori-edge/k8s_gateway/blob/master/kubernetes.go#L189)

and [Handle Error](https://github.com/ori-edge/k8s_gateway/blob/master/kubernetes.go#L201)

currently, if gateway crds aren't installed, it'll just skip over the resources completely, which is what we want.

Reply:
ah, I see.